vmm/src/config.rs

```diff
@@ -2157,12 +2192,97 @@ impl RestoreConfig {
             .map_err(Error::ParseRestore)?
             .unwrap_or(Toggle(false))
             .0;
+        let net_fds = parser
+            .convert::<Tuple<String, Vec<u64>>>("net_fds")
+            .map_err(Error::ParseRestore)?
+            .map(|v| {
+                v.0.iter()
+                    .map(|(id, fds)| RestoredNetConfig {
+                        id: id.clone(),
+                        num_fds: fds.len(),
+                        fds: Some(fds.iter().map(|e| *e as i32).collect()),
+                    })
+                    .collect()
+            });
 
         Ok(RestoreConfig {
             source_url,
             prefault,
+            net_fds,
         })
     }
+
+    pub fn validate(&self, vm_config: &VmConfig) -> ValidationResult<()> {
```

Comment:
The validation logic here is a bit messy, and should be simplified.

Reply:
@likebreath Thanks for the commit to simplify validation logic.  Highlighting the new changes made.
Squashed some commits as suggested.
Removed the refactoring commit in http_endpoint.rs.
Added `sum(num_fds) == files.len()` check in Puthandler and returned `HttpError::BadRequest` if not equal.